cloudbridge/cloud/providers/gce/services.py

```diff
@@ -797,14 +799,28 @@ def create(self, network, cidr_block, name=None, zone=None):
                 return None
             subnets = self.list(network, region)
             for subnet in subnets:
-                cb.log.warning('subnet ID: %s', subnet.id)
                 if subnet.id == response['targetId']:
                     return subnet
         except:
             return None
 
     def get_or_create_default(self, zone=None):
-        raise NotImplementedError('To be implemented')
+        """
+        Every GCP project comes with a default auto mode VPC network. An auto
+        mode VPC network has exactly one subnetwork per region. This method
+        returns the subnetwork of the default network that spans the given
+        zone.
+        """
+        network = self.provider.network.get_by_name(
```

Comment:
This should probably be replaced by self.provider.networking.networks.find(name=GCEFirewallsDelegate.DEFAULT_NETWORK)[0]

Also, in some recent refactoring we did, networks have been moved under networking for greater consistency, and the parameter order for network.create() has been changed.

Which also brings up the question, do we need the get_by_name function since the find function exists?

Reply:
N/A: this part is removed in the newer version #74.